src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -262,32 +264,57 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
                 {
                     TraceData[i].GetObjectData(info, context, i, "Base-");
                 }
+            }
 
+            internal static TraceSerialiserHelper FromCallSiteData(string callSiteData)
```

Comment:
I created this new static constructor for `TraceSerializerHelper` from existing logic.